### PR TITLE
Patch/static analysis

### PR DESCRIPTION
**Pull request by johnmay**
Patches addressing #523. I've skipped SMSD ones for now see that code is deprecated so meh.

### CODE REVIEW

File: descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java [hunk 1]
```diff
@@ -331,20 +331,19 @@ public double extractIP(IAtomContainer container, IAtom atom) {
...
+        StringBuilder idEdited  = new StringBuilder();
+        StringBuilder valEdited = new StringBuilder();
 
         int strlen = str.length();
 
         boolean foundSpace = false;
-        int countSpace = 0;
         boolean foundDigit = false;
         for (int i = 0; i < strlen; i++) {
-            if (!foundDigit) if (Character.isLetter(str.charAt(i))) foundDigit = true;
```
Comment (egonw) [inline descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java]: Not changed in your patch, so no worries, but I really don't understand what this code is supposed to do... not very useful variable naming :(

Comment (johnmay) [reply descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java]: Well that's the part that's failing.... :-) it's basically trying to split the input into <hose_code><space><id>

Comment (egonw) [reply descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java]: It was failing but fixed now, or was failing and still is? This line you did not change, just reformat, right? Or am I blind. My point was `foundDigit` suggests it's looking for digits... but isLetter() is not doing that, or am I missing something?

Comment (johnmay) [reply descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java]: No not changed yet, currently failing.

**Comment by johnmay** [comment]
okay updated

**Pull request merged**